framework/include/utils/MooseUtils.h

```diff
@@ -117,6 +118,11 @@ void escape(std::string & str);
  */
 std::string trim(const std::string & str, const std::string & white_space = " \t\n\v\f\r");
 
+/**
+ * Removes additional whitespace from a string
```

Comment:
From the beginning, the end and turning repeated WS into single WS? Did I read the regex right?

Reply:
Correct. I'll add to the comment